pkg/test-infra/cdc/operation.go

```diff
@@ -16,193 +16,225 @@ package cdc
 import (
 	"context"
 	"fmt"
+	"time"
 
 	_ "github.com/go-sql-driver/mysql" // mysql driver
+	"github.com/ngaut/log"
 	"github.com/pingcap/errors"
-	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
 
-	"github.com/pingcap/tipocket/pkg/test-infra/fixture"
+	clusterTypes "github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/test-infra/tidb"
 	"github.com/pingcap/tipocket/pkg/test-infra/util"
 
 	appsv1 "k8s.io/api/apps/v1"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/utils/pointer"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-// CdcOps knows how to operate TiDB CDC on k8s
-type CdcOps struct {
+// CDCOps knows how to operate TiDB CDC on k8s
+type CDCOps struct {
 	cli client.Client
+	*tidb.TidbOps
 }
 
-func New(cli client.Client) *CdcOps {
-	return &CdcOps{cli}
+// New creates cdc ops
+func New(cli client.Client, tidbClient *tidb.TidbOps) *CDCOps {
+	return &CDCOps{cli, tidbClient}
 }
 
-type CDCSpec struct {
-	Namespace string
-	Name      string
-	Source    *v1alpha1.TidbCluster
-	Replicas  int32
-	Image     string
-	Resources corev1.ResourceRequirements
-}
+// Apply CDC cluster
+func (c *CDCOps) Apply(tc *Recommendation) error {
+	if err := c.ApplyTiDBCluster(tc.Upstream); err != nil {
+		return err
+	}
+
+	if err := c.ApplyTiDBCluster(tc.Downstream); err != nil {
+		return err
+	}
+
+	if err := c.ApplyCDC(tc.CDC); err != nil {
+		return err
+	}
+
+	if err := c.applyJob(tc.CDC.Job); err != nil {
+		return err
+	}
 
-type CDC struct {
-	Sts    *appsv1.StatefulSet
-	Source *v1alpha1.TidbCluster
+	time.Sleep(2 * time.Second)
```

Comment:
Why sleep here? Could you please add some comments?

Reply:
this is unnecessary, I will remove it.